src/background.js

```diff
@@ -34,158 +36,394 @@
+      for(var i = 0; i < tabs.length; i++){
+        var tab = tabs[i];
+        pb.tabData[tab.id] = {
+          frames: {
+            0: {
+              parent: -1,
+              url: tab.url
+            }
+          },
+          trackers: {}
+        };
+      }
+    });
+  },
+
+  /**
+   * Initialize the Cookieblock List:
+   * * Download list form eff
+   * * Merge with existing cookieblock list if any
+   * * Add any new domains to the action map
+   * Set a timer to update every 24 hours
+   **/
+  initializeCookieBlockList: function(){
+    pb.updateCookieBlockList();
+    setInterval(pb.updateCookieBlockList, pb.utils.oneDay());
+  },
+
+  /**
+  * Update the cookie block list with a new list
+  * add any new entries that already have a parent domain in the action_map
+  * and remove any old entries that are no longer in the cookie block list
+  * from the action map
+  **/
+  updateCookieBlockList: function(){
+    pb.utils.xhrRequest(pb.COOKIE_BLOCK_LIST_URL, function(err,response){
+      if(err){
+        console.error('Problem fetching privacy badger policy hash list at',
+                  pb.COOKIE_BLOCK_LIST_URL, err.status, err.message);
+        return;
+      }
+      var cookieblock_list = pb.storage.getBadgerStorageObject('cookieblock_list');
+      var action_map = pb.storage.getBadgerStorageObject('action_map');
+
+      var newCbDomains = _.map(response.split("\n"), function(d){ return d.trim();});
+      var oldCbDomains = Object.keys(cookieblock_list.getItemClones());
+
+      var addedDomains = _.difference(newCbDomains, oldCbDomains);
+      var removedDomains = _.difference(oldCbDomains, newCbDomains);
+      pb.log('adding to cookie blocklist:', addedDomains);
+      pb.log('removing from cookie blocklist:', removedDomains);
+
+      // Change any removed domains back to blocked status
+      _.each(removedDomains, function(domain){
+        cookieblock_list.deleteItem(domain);
+        if(action_map.hasItem(domain)){
+          pb.storage.setupHeuristicAction(domain, pb.BLOCK);
+        }
+        var rmvdSubdomains = _.filter(Object.keys(action_map.getItemClones()),
+                                  function(subdomain){
+                                    return subdomain.endsWith(domain);
+                                  });
+        _.each(removedDomains, function(domain){
+          pb.storage.setupHeuristicAction(domain, pb.BLOCK);
+        });
+      });
+
+      // Add any new cookie block domains who's parent domain is already blocked
+      _.each(addedDomains, function(domain){
+        cookieblock_list.setItem(domain, true);
+        var baseDomain = window.getBaseDomain(domain);
+        if(action_map.hasItem(baseDomain) &&
+           _.contains([pb.BLOCK, pb.COOKIEBLOCK],
+                      action_map.getItem(baseDomain).heuristicAction)){
+          pb.storage.setupHeuristicAction(domain, pb.COOKIEBLOCK);
```

Comment:
so only things that are blocked get cookie blocked
so the flow is that if a domain isn't seen, we wait til it gets blocked and then the next day the cookie block list fetch sets it to cookieblock?